Home_task_shop.py

```diff
@@ -0,0 +1,165 @@
+class Product:
+    def __init__(self, name: str, price: int, unit: float, _purchased):
+        self.name = name
+        self.price = price
+        self.unit = unit
+        self._purchased = True
+
+    def __str__(self) -> str:
+        return self.name
+
+    def __float__(self) -> float:
+        return self.get_total() / 100
+
+    def __eq__(self, other) -> bool:
+        return self.name == other.name and self.price == other.price and self.unit == other.unit
+
+    def get_total(self, quantity: int | float = None):
+        if quantity:
+            return int(round(quantity * self.price / self.unit, 0))
+        else:
+            return self.price
+
+
+class ShoppingCart:
+    def __init__(self):
+        self.products = []
+        self.quantities = []
+        self.index = 0
+
+    def __float__(self) -> float:
+        return self.get_total() / 100
+
+    def __eq__(self, other) -> bool:
+        return self.products == other.products and self.quantities == other.quantities
+
+    def add_product(self, product: Product, quantity: int | float = None) -> None:
+        quantity = quantity or product.unit
+        if product not in self.products:
+            self.products.append(product)
+            self.quantities.append(quantity)
+        else:
+            index_product = self.products.index(product)
+            self.quantities[index_product] += quantity
+
+    def __getitem__(self, key) -> tuple:
+        return self.products[key], self.quantities[key]
+
+    def __len__(self) -> int:
+        return len(self.products)
+
+    def __iter__(self):
+        return zip(self.products, self.quantities)
+
+    def remove_product(self, product: Product) -> None:
+        if product in self.products:
+            index_product = self.products.index(product)
+            self.products.pop(index_product)
+            self.quantities.pop(index_product)
+        else:
+            print('Sorry, we have not such product')
+
+    def sub_product(self, product: Product, quantity: int | float = None):
+        index_product = self.products.index(product)
+        self.quantities[index_product] -= quantity
+        if self.quantities[index_product] <= 0:
+            self.remove_product(product)
+        else:
+            print('Зменшили суму')
+
+    def get_total(self):
+        return sum(product.get_total(quantity) for product, quantity in ShoppingCart.__iter__(self))
+
+
+class PaymentValidator:
+    def is_valid(self):
+        return True if self.is_valid else False
+
+
+class PaymentProcessor:
+    def purchase(self):
+        'XXXXXX'
```

Comment:
Не помітив минулого разу
```suggestion
        raise NotImplementedError
```